### ISSUE TITLE
Kubermatic Installer Does not Update Dex Configuration

### ISSUE THREAD

**Issue by ag1989** [kubermatic/kubermatic#6105]
**User Story**

When trying to update the DEX configuration (e.g. adding an additional IdP) through the kubermatic-installer's `apply` option, no changes are applied to DEX. Using the `--force` flag changes the configuration but breaks certs issued by https://letsencrypt.org. This was observed with the community edition, cluster was bootstrapped by kubeone, Nginx Loadbalancer in front of nodes with ingress-controlers, cluster worked perfectly before trying to make changes :D. Manually changing the DEX configuration through the according HELM chart is not working as HELM complains about a non existing release. 

**Acceptance criteria**

- Changed configurations in the `values.yml` should be applied when running `./kubermatic-installer apply`

**Comment by xrstf** [comment]
Hi there!

> Using the --force flag changes the configuration but breaks certs issued by https://letsencrypt.org. 

Can you explain a bit more as to what happened there? Updating Dex/IDP itself should not break certificates, just extend the existing certs to include more or fewer domains.

> Manually changing the DEX configuration through the according HELM chart is not working as HELM complains about a non existing release.

That should not be the case. What Helm version did you use? How did you try to update chart? You can also run the installer in verbose mode, it will then print the exact Helm commands it uses.

**Comment by ag1989** [comment]
Hi @xrstf 

> Can you explain a bit more as to what happened there? Updating Dex/IDP itself should not break certificates, just extend the existing certs to include more or fewer domains

Sure! We had a running Kubermatic installation (2.15) and wanted to make some changes to the used IdPs we are using (adding a LDAP one). When I run the installer again, we did not see any changes to be applied to DEX. However, our cert seemed to be broken as all browser marked it as insecure, although it worked with any issues before. While the CERT issue might be linked to using the wrong issuer, no updates to DEX seemed more like an issue to me.

> That should not be the case. What Helm version did you use? How did you try to update chart? You can also run the installer in verbose mode, it will then print the exact Helm commands it uses.

I added the according section in my 'values.yml' file and rerun the installer with the new `values.yml`. According to the cli logs everything went well, however, no changes were applied. As the cluster has been taken down and reinstalled due to several other changes we made (its a demo system atm) I cannot reproduce the issue. In case I need to make similar changes to the current demo cluster and experience the same issue I will try to provide more insights.
